en/tutorials/camera_trigger.md

```diff
@@ -1,86 +1,129 @@
-* `TRIG_INTERFACE` 4 enables the generic PWM interface. This allows the use of  [infrared triggers](http://www.himodel.com/FPV_Telemetry/Super_Tiny_1g_Only_Universal_Remote_Control_Infrared_Shutter_IR-RC.html) or servos to trigger your camera.
+## Trigger Interface Backends
 
-### Other parameters 
+The camera trigger driver supports several backends for triggering different types of camera interfaces. The type of camera can be specified using the [TRIG_INTERFACE](../advanced/parameter_reference.md#TRIG_INTERFACE) parameter: 
 
-* `TRIG_POLARITY` - Relevant only while using the GPIO interface. Sets the polarity of the trigger pin. Active high means that the pin is pulled low normally and pulled high on a trigger event. Active low is vice-versa.
-* `TRIG_INTERVAL` - Defines the time between two consecutive trigger events in milliseconds.
-* `TRIG_ACTIVATION_TIME`- Defines the time in milliseconds the trigger pin is held in the "active" state before returning to neutral. In PWM modes, the minimum is limited to 40 ms to make sure we always fit an activate pulse into the 50Hz PWM signal.
+Value | Interface | Description
+--- | --- | ---
+`1` | *GPIO interface* | The AUX outputs are pulsed high or low (depending on the `TRIG_POLARITY` parameter) every `TRIG_INTERVAL` duration. This can be used to trigger most standard machine vision cameras directly. Note that on PX4FMU series hardware (Pixhawk, Pixracer, etc.), the signal level on the AUX pins is 3.3v.
+`2` | *Seagull MAP2 interface* | This allows the use of the [Seagull MAP2](http://www.seagulluav.com/product/seagull-map2/) to interface to a multitude of supported cameras. Pin 1 of the MAP2 should be connected to the lower AUX pin of `TRIG_PINS` (therefore, pin 1 to AUX 5 and pin 2 to AUX 6 by default). In this mode, PX4 also supports automatic power control and keep-alive functionalities of *Sony Multiport* cameras like the QX-1.
+`3` | *MAVLink interface* | In this mode, no actual hardware output is used - the driver only sends the [CAMERA_TRIGGER](https://mavlink.io/en/messages/common.html#CAMERA_TRIGGER) MAVLink message (by default, if the MAVLink application is in `onboard` mode. Otherwise, a custom stream will need to be enabled). PX4 must separately forward all MAVLink camera messages from a GCS to a connected MAVLink camera.
+`4` | Generic PWM interface | Allows the use of  [infrared triggers](http://www.himodel.com/FPV_Telemetry/Super_Tiny_1g_Only_Universal_Remote_Control_Infrared_Shutter_IR-RC.html) or servos to trigger your camera.
+
+### Other Parameters 
+
+* [TRIG_POLARITY](../advanced/parameter_reference.md#TRIG_POLARITY) - Relevant only while using the GPIO interface. Sets the polarity of the trigger pin. Active high means that the pin is pulled low normally and pulled high on a trigger event. Active low is vice-versa.
+* [TRIG_INTERVAL](../advanced/parameter_reference.md#TRIG_INTERVAL) - Defines the time between two consecutive trigger events in milliseconds.
```

Comment:
This appears to only be read when the driver is initialised. The value used can be changed via mavlink messages but the parameter itself is not re-read. Is it a problem that this is not updated again if the parameter is changed after the driver has started?

Perhaps add note that device must be restarted for this change to take effect?

Reply:
@ndepal Can you comment on this?

Reply:
I don't have any experience with camera triggering. I read through this PR because I was curious about the feature.

That being said, it appears to me that the `camera_trigger` actually handles those mavlink messages (and writes the param to store the new value premanently): https://github.com/PX4/Firmware/blob/master/src/drivers/camera_trigger/camera_trigger.cpp#L619

So my guess is that no restart is required.

Reply:
You're right. Thank you!